code/modules/cciaa/cciaa_items.dm

```diff
@@ -107,88 +53,109 @@
 		to_chat(usr, "<span class='notice'>The device beeps and flashes \"Already recording, Aborting\".</span>")
 		return
 
-	var/case = input(usr, "Enter case name","Case Name") as null|text
-	if(!case || case == "")
-		to_chat(usr, "<span class='notice'>The device beeps and flashes \"No data entered, Aborting\".</span>")
+	//If nothing has been done with the device yet
+	if(!report_id && !interviewee_id)
+		if(config.sql_ccia_logs)
+			//Get the active cases from the database and display them
+			var/list/reports = list()
+			var/DBQuery/report_query = dbcon.NewQuery("SELECT id, report_date, title FROM ss13_ccia_reports WHERE status = 'approved' AND deleted_at IS NULL")
+			report_query.Execute()
+			while(report_query.NextRow())
+				reports["[report_query.item[2]] - [report_query.item[3]]"] = "[report_query.item[1]]"
```

Comment:
This basically pulls all. Might want to pagify or, at minimum, tick-check this with `CHECK_TICK`

Reply:
Well, that pulls all active IRs, so that are around 5-10 at most